kie-internal/src/main/java/org/kie/internal/builder/conf/AlphaNetworkCompilerOption.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2010 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.internal.builder.conf;
+
+/**
+ * An Enum for AlphaNetworkCompilerOption option.
+ *
+ * drools.alphaNetworkCompiler = &lt;disabled|inmemory|compiled&gt;
+ *
+ * Disabled: do not generate compiled alpha network
+ * InMemory: generate compiled alpha network after creation of the kiebase and compile it in-memory
+ * Compiled: assume compiled alpha network is already compiled in the kjar. To be used with the executable model
+ *
+ * DEFAULT = disabled
+ */
+public enum AlphaNetworkCompilerOption implements SingleValueKnowledgeBuilderOption {
+
+    DISABLED("disabled"),
+    INMEMORY("inmemory"),
+    COMPILE("compile"),
+    LOAD("load");
```

Comment:
@lucamolteni can you please clarify (and maybe add a comment line in javadocs) what is this LOAD option for?